Framework/Core/src/CompletionPolicyHelpers.cxx

```diff
@@ -198,13 +198,21 @@ CompletionPolicy CompletionPolicyHelpers::consumeExistingWhenAny(const char* nam
 
 CompletionPolicy CompletionPolicyHelpers::consumeWhenAny(const char* name, CompletionPolicy::Matcher matcher)
 {
-  auto callback = [](InputSpan const& inputs) -> CompletionPolicy::CompletionOp {
-    for (auto& input : inputs) {
-      if (input.header != nullptr) {
-        return CompletionPolicy::CompletionOp::Consume;
+  auto callback = [](InputSpan const& inputs, std::vector<InputSpec> const& specs) -> CompletionPolicy::CompletionOp {
```

Comment:
@ktf Actually, I see that this callback is never called when the policy is defined via 
```
CompletionPolicyHelpers::defineByName(<device_name>, CompletionPolicy::CompletionOp::Consume))
```
It is called only if I substitute https://github.com/AliceO2Group/AliceO2/blob/5a46e001502eb4acd5da6a174b2d725d1ca6e301/Framework/Core/src/CompletionPolicyHelpers.cxx#L83 by 
```
return consumeWhenAny(name.c_str(), matcher);
```
But then there is another problem: for some reason, the condition specs are always in the very end of the specs vector and they are also received as the last input (does the order in which the inputs are received depend on their position in the InputSpec vector?), which effectively invalidates the `ConsumeWhenAny` policy. I've tried to sort the InputSpecs vector in the device definition by 
```
auto conditionsFirst = [](std::vector<o2::framework::InputSpec>& inputs) 
{                                                                                                                                                                                                                                        
    std::sort(inputs.begin(), inputs.end(), [](const o2::framework::InputSpec& a, const o2::framework::InputSpec& b) { return a.lifetime==o2::framework::Lifetime::Condition && b.lifetime!=o2::framework::Lifetime::Condition; });
}; 
```
and when I print it in the `getTPCDigitizerSpec` it shows the order I want:
```
[2100238:TPCDigitizer]: [INFO] spec 0 : TPC/VDriftTgl/0
[2100238:TPCDigitizer]: [INFO] spec 1 : TPC/CalibLaserTracks/0
[2100238:TPCDigitizer]: [INFO] spec 2 : SIM/COLLISIONCONTEXT/20
...
[2100238:TPCDigitizer]: [INFO] spec 37 : SIM/COLLISIONCONTEXT/18
```
but then, when I print the input specs from this `consumeWhenAny` I see 
```
[2100238:TPCDigitizer]: [00:31:40][INFO] spec 0 : SIM/COLLISIONCONTEXT/26
[2100238:TPCDigitizer]: [00:31:40][INFO] spec 1 : SIM/COLLISIONCONTEXT/0
...
[2100238:TPCDigitizer]: [00:31:40][INFO] spec 35 : SIM/COLLISIONCONTEXT/25
[2100238:TPCDigitizer]: [00:31:40][INFO] spec 36 : TPC/CalibLaserTracks/0
[2100238:TPCDigitizer]: [00:31:40][INFO] spec 37 : TPC/VDriftTgl/0
```
apparently, you are resorting it when creating the real device. Is there a way to prioritize the conditions?